.dotnet/src/Custom/Chat/ChatResponseFormat.cs

```diff
@@ -1,45 +1,71 @@
-using OpenAI.Models;
+using OpenAI.Internal;
+using System;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
 
 namespace OpenAI.Chat;
 
-/// <summary>
-/// Represents a requested <c>response_format</c> for the model to use, enabling "JSON mode" for guaranteed valid output.
-/// </summary>
-/// <remarks>
-/// <b>Important</b>: when using JSON mode, the model <b><u>must</u></b> also be instructed to produce JSON via a
-/// <c>system</c> or <c>user</c> message.
-/// <para>
-/// Without this paired, message-based accompaniment, the model may generate an unending stream of whitespace until the
-/// generation reaches the token limit, resulting in a long-running and seemingly "stuck" request.
-/// </para>
-/// <para>
-/// Also note that the message content may be partially cut off if <c>finish_reason</c> is <c>length</c>, which
-/// indicates that the generation exceeded <c>max_tokens</c> or the conversation exceeded the max context length for
-/// the model.
-/// </para>
-/// </remarks>
-[CodeGenModel("CreateChatCompletionRequestResponseFormat")]
-public partial class ChatResponseFormat
+[CodeGenModel("ChatResponseFormat")]
+public abstract partial class ChatResponseFormat
 {
-    // CUSTOM: Made internal.
+    public static ChatResponseFormat Text { get; } = new InternalChatResponseFormatText();
+    public static ChatResponseFormat JsonObject { get; } = new InternalChatResponseFormatJsonObject();
 
-    /// <summary> Must be one of `text` or `json_object`. </summary>
-    [CodeGenMember("Type")]
-    internal InternalCreateChatCompletionRequestResponseFormatType? Type { get; set; }
-
-    // CUSTOM: Made internal.
-    /// <summary> Initializes a new instance of <see cref="ChatResponseFormat"/>. </summary>
-    internal ChatResponseFormat()
+    public static ChatResponseFormat CreateTextFormat() => new InternalChatResponseFormatText();
+    public static ChatResponseFormat CreateJsonObjectFormat() => new InternalChatResponseFormatJsonObject();
+    public static ChatResponseFormat CreateJsonSchemaFormat(
+        string name,
+        BinaryData jsonSchema,
+        string description = null,
+        bool? requireStrictJsonSchemaMatch = null)
     {
+        Argument.AssertNotNullOrEmpty(name, nameof(name));
+        Argument.AssertNotNull(jsonSchema, nameof(jsonSchema));
+
+        InternalResponseFormatJsonSchemaJsonSchema internalSchema = new(
+            description,
+            name,
+            jsonSchema,
+            requireStrictJsonSchemaMatch,
+            null);
+        return new InternalChatResponseFormatJsonSchema(internalSchema);
     }
 
-    internal ChatResponseFormat(InternalCreateChatCompletionRequestResponseFormatType? type)
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static bool operator ==(ChatResponseFormat first, ChatResponseFormat second)
+        => first?.Equals(second) ?? false;
```

Comment:
If both are `null`, I think this would incorrectly return `false`.

Reply:
It doesn't -- I'll add test coverage for that and a few other null combinations explicitly.

- if first is `null`, `first?` evaluates to null and coalesces to `false`
- if first is not `null` and `second` is `null`, none of the `Equals` conditions are met and `false` is returned.

I'm all for standardizing this pattern further (and eliminating hand-written parts as much as we can) but I'd like to defer it to a widespread replacement on all of our custom implementations -- we could certainly use some consolidation for that!!

Reply:
Updated one more time -- the null comparison logic is now lifted from the sample for a better match (like `null == null`).